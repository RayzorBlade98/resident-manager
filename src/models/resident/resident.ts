import Name from '../name';
import { RentInformation } from './rent';
import WaterMeterReading from './water_meter_reading';
import MonthYear from '_/extensions/date/month_year.extension';

/**
 * Object containing information about a specific resident
 */
export interface Resident {
  /**
   * Unique id that is used to identify a resident
   */
  id: string;

  /**
   * Name of the resident
   */
  name: Name;

  /**
   * Number of residents living in the appartment
   */
  numberOfResidents: number;

  /**
   * Information about the rent payments
   */
  rentInformation: RentInformation[];

  /**
   * List of all water meter readings
   */
  waterMeterReadings: WaterMeterReading[];

  /**
   * First month and year of the rental contract
   */
  contractStart: MonthYear;

  /**
   * Id of the apartment the resident lives in
   */
  apartmentId: string;

  /**
   * Information about the keys the resident has
   */
  keys: {
    /**
     * Number of apartment keys the resident has
     */
    apartment: number;

    /**
     * Number of basement keys the resident has
     */
    basement: number;

    /**
     * Number of attic keys the resident has
     */
    attic: number;

    /**
     * Number of front door keys the resident has
     */
    frontDoor: number;

    /**
     * Number of mailbox keys the resident has
     */
    mailbox: number;
  }
}
