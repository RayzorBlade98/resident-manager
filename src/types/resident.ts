import { v4 as uuid } from 'uuid';
import {
  createValidationFunction,
  ValidationError,
  ValidationErrorMessages,
} from '../utils/validation';

export interface Resident {
  id: string;
  firstName: string;
  lastName: string;
}

export interface CreateResidentArguments {
  firstName: string;
  lastName: string;
}

export type CreateResidentErrors =
  ValidationErrorMessages<CreateResidentArguments>;

export function createResident(args: CreateResidentArguments): Resident {
  return {
    id: uuid(),
    ...args,
  };
}

export const validateResidentArgs =
  createValidationFunction<CreateResidentArguments>({
    firstName: [ValidationError.EmptyString],
    lastName: [ValidationError.EmptyString],
  });
