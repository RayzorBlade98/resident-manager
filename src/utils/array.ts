/**
 * This module provides functionality regarding `Array` objects
 * like creating the cartesian product of multiple arrays.
 */

/**
 * Creates an array containing a sequence of numbers for a given range.
 *
 * Examples:
 * - `range(0, 4) := [0, 1, 2, 3, 4]`
 * - `range (0, 4, 2) := [0, 2, 4]`
 * - `range (0, 5, 2) := [0, 2, 4]`
 *
 * @param start start value of the sequence
 * @param stop end value of the sequence
 * @param step space between each element of the sequence (default: 1)
 * @returns sequence of numbers for the given range
 */
export function range(start: number, stop: number, step: number = 1): number[] {
  return [...Array(Math.floor((stop - start) / step) + 1)].map(
    (_, i) => start + i * step,
  );
}

/**
 * Creates the cartesian product for any number of given array.
 *
 * Examples:
 * - `cartesianProduct([1, 2], [3, 4]) := [[1, 3], [1, 4], [2, 3], [2, 4]]`
 * - `cartesianProduct([1, 2], [3, 4], [5, 6]) := [[1, 3, 5], [1, 3, 6], [1, 4, 5], [1, 4, 6], [2, 3, 5], [2, 3, 6], [2, 4, 5], [2, 4, 6]]`
 *
 * @param allEntries any number of array that should be combined
 * @returns array containing the cartesian product over all given arrays
 */
export function cartesianProduct<T>(...allEntries: T[][]): T[][] {
  return allEntries.reduce<T[][]>(
    (results, entries) =>
      results
        .map((result) => entries.map((entry) => result.concat([entry])))
        .reduce((subResults, result) => subResults.concat(result), []),
    [[]],
  );
}
