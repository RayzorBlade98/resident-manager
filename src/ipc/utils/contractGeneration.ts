import {
  convertAddressToCityString,
  convertAddressToStreetString,
} from '../../utils/address/address.utils';
import { convertNameToString } from '../../utils/name/name.utils';
import contractTemplate from '_/assets/contract/contractTemplate.md';
import landlordCompanyTemplate from '_/assets/contract/landlordCompanyTemplate.md';
import Landlord from '_/models/landlord/landlord';
import Property from '_/models/property/property';
import { Resident } from '_/models/resident/resident';

export type ContractGenerationArgs = {
  landlord: Landlord;
  resident: Resident;
  property: Property;
};

const placeholderLabels = {
  landlordName: 'LANDLORD_NAME',
  landlordStreet: 'LANDLORD_STREET',
  landlordCity: 'LANDLORD_CITY',
  landlordPhone: 'LANDLORD_PHONE', // Todo
  landlordEmail: 'LANDLORD_EMAIL', // Todo
  landlordCompany: 'LANDLORD_COMPANY',
  residentName: 'RESIDENT_NAME',
  residentStreet: 'RESIDENT_STREET', // Todo
  residentCity: 'RESIDENT_CITY', // Todo
  residentPhone: 'RESIDENT_PHONE', // Todo
  propertyStreet: 'PROPERTY_STREET',
  propertyCity: 'PROPERTY_CITY',
  apartmentFloor: 'APARTMENT_FLOOR',
  apartmentLocation: 'APARTMENT_LOCATION',
  apartmentRoomsGeneric: 'APARTMENT_ROOMS_GENERIC',
  apartmentRoomsKitchen: 'APARTMENT_ROOMS_KITCHEN',
  apartmentRoomsBath: 'APARTMENT_ROOMS_BATH',
  apartmentRoomsHallway: 'APARTMENT_ROOMS_HALLWAY',
  apartmentRoomsBasement: 'APARTMENT_ROOMS_BASEMENT',
  apartmentRoomsGarden: 'APARTMENT_ROOMS_GARDEN', // Todo
  parkingSpaceCount: 'PARKING_SPACE_COUNT',
  keysApartment: 'KEYS_APARTMENT',
  keysBasement: 'KEYS_BASEMENT',
  keysAttic: 'KEYS_ATTIC',
  keysFrontdoor: 'KEYS_FRONTDOOR',
  keysMailbox: 'KEYS_MAILBOX',
} satisfies Record<string, string>;

const blockPlaceholderLabels = {
  landlordCompany: 'LANDLORD_COMPANY_BLOCK',
} satisfies Record<string, string>;

/**
 * Generates the contract as markdown string
 * @returns markdown string of the generated contract
 */
export function generateContractMarkdown(args: ContractGenerationArgs): string {
  const apartment = args.property.apartments.find(
    (a) => a.id === args.resident.apartmentId,
  )!;

  const placeholders = {
    [placeholderLabels.landlordName]: convertNameToString(
      args.landlord.representative,
    ),
    [placeholderLabels.landlordStreet]: convertAddressToStreetString(
      args.landlord.address,
    ),
    [placeholderLabels.landlordCity]: convertAddressToCityString(
      args.landlord.address,
    ),
    [placeholderLabels.landlordCompany]: args.landlord.company ?? '',
    [placeholderLabels.residentName]: convertNameToString(args.resident.name),
    [placeholderLabels.propertyStreet]: convertAddressToStreetString(
      args.property.address,
    ),
    [placeholderLabels.propertyCity]: convertAddressToCityString(
      args.property.address,
    ),
    [placeholderLabels.apartmentFloor]: apartment.floor,
    [placeholderLabels.apartmentLocation]: apartment.location,
    [placeholderLabels.apartmentRoomsGeneric]:
      apartment.rooms.generic.toString(),
    [placeholderLabels.apartmentRoomsKitchen]:
      apartment.rooms.kitchen.toString(),
    [placeholderLabels.apartmentRoomsBath]: apartment.rooms.bath.toString(),
    [placeholderLabels.apartmentRoomsBasement]:
      apartment.rooms.basement.toString(),
    [placeholderLabels.apartmentRoomsHallway]:
      apartment.rooms.hallway.toString(),
    [placeholderLabels.parkingSpaceCount]: args.resident.parkingSpaceId ? '1' : '0',
    [placeholderLabels.keysApartment]: args.resident.keys.apartment.toString(),
    [placeholderLabels.keysBasement]: args.resident.keys.basement.toString(),
    [placeholderLabels.keysAttic]: args.resident.keys.attic.toString(),
    [placeholderLabels.keysFrontdoor]: args.resident.keys.frontDoor.toString(),
    [placeholderLabels.keysMailbox]: args.resident.keys.mailbox.toString(),
  };

  let contract = contractTemplate;

  // Landlord company
  contract = contract.replace(
    getPlaceholder(blockPlaceholderLabels.landlordCompany),
    args.landlord.company ? landlordCompanyTemplate : '',
  );

  return replaceAllPlaceholders(contract, placeholders);
}

function replaceAllPlaceholders(
  contract: string,
  replacements: Record<string, string>,
): string {
  Object.entries(replacements).forEach(([label, replacement]) => {
    contract = contract.replace(getPlaceholder(label), replacement);
  });
  return contract;
}

function getPlaceholder(label: string): string {
  return `{{${label}}}`;
}
