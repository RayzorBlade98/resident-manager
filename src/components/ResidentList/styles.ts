import { css, Rule, StyleAttribute } from "glamor";
import { defaultColors } from "_/types/styles";

const residentListElementStyle: Rule = {
  width: "100%",
  color: defaultColors.mainDark2,
  display: "flex",
  justifyContent: "center",
  alignItems: "center",
  ":hover": {
    backgroundColor: defaultColors.mainLight3,
    cursor: "pointer",
  },
};

const residentListContainer: Rule = {
  width: "20%",
  height: "100%",
  backgroundColor: defaultColors.mainLight2,
  overflowY: "auto",
};

const newResidentDiv: Rule = { ...residentListElementStyle, height: "5%" };

const residentListElementContainer: Rule = {
  ...residentListElementStyle,
  height: "10%",
};

const styles: { [key: string]: StyleAttribute } = {
  residentListContainer: css(residentListContainer),
  newResidentDiv: css(newResidentDiv),
  residentListElementContainer: css(residentListElementContainer),
};

export default styles;
