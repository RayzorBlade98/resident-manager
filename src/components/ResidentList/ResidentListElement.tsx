/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import styles from './styles';
import { Resident } from '_/types/resident';

interface ResidentListElementProps {
  resident: Resident;
  selected: boolean;
  onSelectResident: (resident: Resident) => void;
}

function ResidentListElement(props: ResidentListElementProps): JSX.Element {
  return (
    <div
      onClick={() => {
        props.onSelectResident(props.resident);
      }}
      {...(props.selected
        ? styles.residentListElementContainerSelected
        : styles.residentListElementContainer)}
    >
      {`${props.resident.firstName} ${props.resident.lastName}`}
    </div>
  );
}

export default ResidentListElement;
