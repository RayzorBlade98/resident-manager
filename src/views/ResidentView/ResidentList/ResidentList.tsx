import React, { useState } from 'react';
import { useRecoilState, useRecoilValue } from 'recoil';
// eslint-disable-next-line max-len
import { residentViewSelectedResidentState } from '../states/resident_view_state';
import styles from '../styles';
// eslint-disable-next-line max-len
import GenericList from '_/components/GenericComponents/GenericList/GenericList';
// eslint-disable-next-line max-len
import GenericListElement from '_/components/GenericComponents/GenericList/GenericListElement';
import residentState from '_/states/saveStates/resident_state';
import { Resident } from '_/types/resident';
// eslint-disable-next-line max-len
import CreateResidentModal from '_/views/ResidentView/ResidentList/CreateResidentModal/CreateResidentModal';

/**
 * Component that displays a list of provided residents
 */
function ResidentList(): JSX.Element {
  const residents = useRecoilValue(residentState);
  const [selectedResident, setSelectedResident] = useRecoilState(
    residentViewSelectedResidentState,
  );
  const [showModal, setShowModal] = useState<boolean>(false);

  return (
    <>
      {showModal && (
        <CreateResidentModal
          show={showModal}
          onClose={() => setShowModal(false)}
        />
      )}
      <GenericList style={styles.residentList.container}>
        <GenericListElement
          onClick={() => setShowModal(true)}
          style={styles.residentList.newResident}
        >
          Neuer Mieter
        </GenericListElement>
        {residents.map((resident: Resident) => (
          <GenericListElement
            onClick={() => setSelectedResident(resident)}
            selected={resident === selectedResident}
            key={resident.id}
          >
            {`${resident.firstName} ${resident.lastName}`}
          </GenericListElement>
        ))}
      </GenericList>
    </>
  );
}

export default ResidentList;
