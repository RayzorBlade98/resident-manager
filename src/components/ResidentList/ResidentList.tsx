/* eslint-disable react/jsx-props-no-spreading */
import { StyleAttribute } from 'glamor';
import React, { useState } from 'react';
import CreateResidentModal from '../CreateResidentModal/CreateResidentModal';
import ResidentListElement from './ResidentListElement';
import styles from './styles';
import { Resident } from '_/types/resident';

interface ResidentListProps {
  residents: Resident[];
  selectedResident: Resident;
  onSelectResident: (resident: Resident) => void;
  containerStyle?: StyleAttribute;
}

function ResidentList(props: ResidentListProps): JSX.Element {
  const [showModal, setShowModal] = useState<boolean>(false);

  function onNewResident(): void {
    setShowModal(true);
  }

  return (
    <>
      {showModal && (
        <CreateResidentModal
          show={showModal}
          onClose={() => setShowModal(false)}
        />
      )}

      <div {...{ ...styles.residentListContainer, ...props.containerStyle }}>
        <div {...styles.newResidentDiv} onClick={onNewResident}>
          Neuer Mieter
        </div>
        <hr className="hr" />
        {props.residents.map((resident: Resident) => (
          <>
            <ResidentListElement
              resident={resident}
              selected={resident.id === props.selectedResident.id}
              onSelectResident={props.onSelectResident}
              key={resident.id}
            />
            {resident.id !== props.residents.slice(-1)[0].id && (
              <hr className="hr" />
            )}
          </>
        ))}
      </div>
    </>
  );
}

export default ResidentList;
