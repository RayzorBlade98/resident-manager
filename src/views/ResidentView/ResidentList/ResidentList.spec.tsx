/* eslint-disable max-len, @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment */
/* eslint-disable @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call, func-names */

import {
  RenderResult,
  render,
  fireEvent,
  act,
  prettyDOM,
} from '@testing-library/react';
import React from 'react';
import { getRecoil, setRecoil } from 'recoil-nexus';
import residentViewState from '../states/resident_view_state';
import ResidentList from './ResidentList';
import residentState from '_/states/saveStates/resident_state';
import { Resident } from '_/types/resident';
import { range } from '_/utils/array';
import RecoilTestWrapper from '_tests/__test_utils__/RecoillTestWrapper';
import ResidentBuilder from '_tests/__test_utils__/builders/resident_builder';

const createResidentModalPropsMock = jest.fn();
jest.mock(
  '_/views/ResidentView/ResidentList/CreateResidentModal/CreateResidentModal',
  () => function (props: any) {
    createResidentModalPropsMock(props);
    return <div className="modalMock" />;
  },
);

describe('ResidentList', () => {
  let renderResult: RenderResult;
  let residents: Resident[];
  let selectedResident: Resident;

  beforeEach(() => {
    residents = range(0, 4).map((i) => new ResidentBuilder()
      .withFirstName(`Max${i}`)
      .withLastName(`Mustermann${i}`)
      .build());
    selectedResident = residents[0];

    renderResult = render(
      <RecoilTestWrapper>
        <ResidentList />
      </RecoilTestWrapper>,
    );

    act(() => {
      setRecoil(residentState, residents);
      setRecoil(residentViewState, (state) => ({
        ...state,
        selectedResident: selectedResident.id,
      }));
    });
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  function selectResident(index: number): void {
    const listElement = renderResult.container.firstElementChild!.children.item(
      1 + 2 * index + 1,
    )!;
    fireEvent.click(listElement);
  }

  function clickNewResident(): void {
    const newResidentElement = renderResult.container.firstElementChild!.children.item(0)!;
    fireEvent.click(newResidentElement);
  }

  test('should have the right amount of children', () => {
    // Assert
    const expectedChildren = 2 * residents.length + 1;
    const actualChildren = renderResult.container.firstElementChild!.children.length;
    expect(actualChildren).toEqual(expectedChildren);
  });

  test('should set selected resident in state when clicking list element', () => {
    // Act
    const selectedIndex = 1;
    selectResident(selectedIndex);

    // Assert
    const newSelectedResident = getRecoil(residentViewState).selectedResident;
    expect(newSelectedResident).toEqual(residents[selectedIndex].id);
  });

  test('should close modal on callback', () => {
    // Arrange
    const oldDOM = prettyDOM(renderResult.container);

    // Act
    clickNewResident();
    const onCloseCallback = createResidentModalPropsMock.mock.calls[0][0].onClose;
    act(() => {
      onCloseCallback();
    });

    // Assert
    expect(prettyDOM(renderResult.container)).toEqual(oldDOM);
  });

  test('should match snapshot', () => {
    // Assert
    expect(renderResult.container).toMatchSnapshot();
  });

  test('should match snapshot (opened modal)', () => {
    // Act
    clickNewResident();

    // Assert
    expect(renderResult.container).toMatchSnapshot();
  });

  test('should match snapshot (new selection)', () => {
    // Act
    selectResident(2);

    // Assert
    expect(renderResult.container).toMatchSnapshot();
  });
});
