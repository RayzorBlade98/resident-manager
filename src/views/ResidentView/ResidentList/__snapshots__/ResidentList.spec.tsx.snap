// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResidentList should match snapshot (new selection) 1`] = `
.css-0,
[data-css-0] {
  width: 100%;
  height: 10%;
  color: #163172;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-0:hover,
[data-css-0]:hover {
  background-color: #c8daea;
  cursor: pointer;
}

.css-2,
[data-css-2] {
  width: 100%;
  height: 10%;
  color: #163172;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #c8daea;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-2:hover,
[data-css-2]:hover {
  background-color: #c8daea;
  cursor: pointer;
}

.css-4,
[data-css-4] {
  background-color: #d6e4f0;
  overflow-y: auto;
  width: 20%;
  height: 100%;
}

.css-5,
[data-css-5] {
  width: 100%;
  height: 5%;
  color: #163172;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-5:hover,
[data-css-5]:hover {
  background-color: #c8daea;
  cursor: pointer;
}

<div>
  <div
    class="genericList"
    data-css-4=""
  >
    <div
      class="listElement"
      data-css-5=""
    >
      Neuer Mieter
    </div>
    <hr
      class="hr"
    />
    <div
      class="listElement"
      data-css-0=""
    >
      Max0 Mustermann0
    </div>
    <hr
      class="hr"
    />
    <div
      class="listElement"
      data-css-0=""
    >
      Max1 Mustermann1
    </div>
    <hr
      class="hr"
    />
    <div
      class="listElement selectedListElement"
      data-css-2=""
    >
      Max2 Mustermann2
    </div>
    <hr
      class="hr"
    />
    <div
      class="listElement"
      data-css-0=""
    >
      Max3 Mustermann3
    </div>
    <hr
      class="hr"
    />
    <div
      class="listElement"
      data-css-0=""
    >
      Max4 Mustermann4
    </div>
  </div>
</div>
`;

exports[`ResidentList should match snapshot (opened modal) 1`] = `
.css-0,
[data-css-0] {
  width: 100%;
  height: 10%;
  color: #163172;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-0:hover,
[data-css-0]:hover {
  background-color: #c8daea;
  cursor: pointer;
}

.css-2,
[data-css-2] {
  width: 100%;
  height: 10%;
  color: #163172;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #c8daea;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-2:hover,
[data-css-2]:hover {
  background-color: #c8daea;
  cursor: pointer;
}

.css-4,
[data-css-4] {
  background-color: #d6e4f0;
  overflow-y: auto;
  width: 20%;
  height: 100%;
}

.css-5,
[data-css-5] {
  width: 100%;
  height: 5%;
  color: #163172;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-5:hover,
[data-css-5]:hover {
  background-color: #c8daea;
  cursor: pointer;
}

<div>
  <div
    class="modalMock"
  />
  <div
    class="genericList"
    data-css-4=""
  >
    <div
      class="listElement"
      data-css-5=""
    >
      Neuer Mieter
    </div>
    <hr
      class="hr"
    />
    <div
      class="listElement selectedListElement"
      data-css-2=""
    >
      Max0 Mustermann0
    </div>
    <hr
      class="hr"
    />
    <div
      class="listElement"
      data-css-0=""
    >
      Max1 Mustermann1
    </div>
    <hr
      class="hr"
    />
    <div
      class="listElement"
      data-css-0=""
    >
      Max2 Mustermann2
    </div>
    <hr
      class="hr"
    />
    <div
      class="listElement"
      data-css-0=""
    >
      Max3 Mustermann3
    </div>
    <hr
      class="hr"
    />
    <div
      class="listElement"
      data-css-0=""
    >
      Max4 Mustermann4
    </div>
  </div>
</div>
`;

exports[`ResidentList should match snapshot 1`] = `
.css-0,
[data-css-0] {
  width: 100%;
  height: 10%;
  color: #163172;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-0:hover,
[data-css-0]:hover {
  background-color: #c8daea;
  cursor: pointer;
}

.css-2,
[data-css-2] {
  width: 100%;
  height: 10%;
  color: #163172;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #c8daea;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-2:hover,
[data-css-2]:hover {
  background-color: #c8daea;
  cursor: pointer;
}

.css-4,
[data-css-4] {
  background-color: #d6e4f0;
  overflow-y: auto;
  width: 20%;
  height: 100%;
}

.css-5,
[data-css-5] {
  width: 100%;
  height: 5%;
  color: #163172;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-5:hover,
[data-css-5]:hover {
  background-color: #c8daea;
  cursor: pointer;
}

<div>
  <div
    class="genericList"
    data-css-4=""
  >
    <div
      class="listElement"
      data-css-5=""
    >
      Neuer Mieter
    </div>
    <hr
      class="hr"
    />
    <div
      class="listElement selectedListElement"
      data-css-2=""
    >
      Max0 Mustermann0
    </div>
    <hr
      class="hr"
    />
    <div
      class="listElement"
      data-css-0=""
    >
      Max1 Mustermann1
    </div>
    <hr
      class="hr"
    />
    <div
      class="listElement"
      data-css-0=""
    >
      Max2 Mustermann2
    </div>
    <hr
      class="hr"
    />
    <div
      class="listElement"
      data-css-0=""
    >
      Max3 Mustermann3
    </div>
    <hr
      class="hr"
    />
    <div
      class="listElement"
      data-css-0=""
    >
      Max4 Mustermann4
    </div>
  </div>
</div>
`;
